### PR TITLE
Add GitHub App authentication

### PR DESCRIPTION
**Pull request by borjalopez-sg**
## Change description

> This feature allows using a GitHub App to authenticate and get a temporal token. This token is configured in the git. Then Terraform is able to use private modules from a GitHub organization.

## Type of change
- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [x] Lint rules pass locally
- [x] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: codebuild-terraform.tf [hunk 1]
```diff
@@ -23,6 +23,31 @@ resource "aws_codebuild_project" "terraform_plan" {
       name  = "TF_IN_AUTOMATION"
       value = "true"
     }
+    dynamic "environment_variable" {
```
Comment (wesleykirklandsg) [inline codebuild-terraform.tf:26]: Please add line breaks for readability between `}` and `dynamic` on each instance in this file

File: README.md [hunk 2]
```diff
@@ -220,6 +220,9 @@ POLICY
 | <a name="input_cb_env_image_pull_credentials_type"></a> [cb\_env\_image\_pull\_credentials\_type](#input\_cb\_env\_image\_pull\_credentials\_type) | The type of credentials AWS CodeBuild uses to pull images in your build. There are two valid values described in [the ProjectEnvironment documentation](https://docs.aws.amazon.com/codebuild/latest/APIReference/API_ProjectEnvironment.html). | `string` | `"CODEBUILD"` | no |
 | <a name="input_cb_env_name"></a> [cb\_env\_name](#input\_cb\_env\_name) | Should be referenced from env\_name of calling terraform module. | `string` | n/a | yes |
 | <a name="input_cb_env_type"></a> [cb\_env\_type](#input\_cb\_env\_type) | Codebuild Environment to use for stages in the pipeline. Valid Values are documented at [the ProjectEnvironment documentation](https://docs.aws.amazon.com/codebuild/latest/APIReference/API_ProjectEnvironment.html). | `string` | `"LINUX_CONTAINER"` | no |
+| <a name="input_cb_gh_app_id"></a> [cb\_gh\_app\_id](#input\_cb\_gh\_app\_id) | ID of the GitHub App. | `string` | `""` | no |
```
Comment (wesleykirklandsg) [inline README.md:223]: The readme should have example files in the documentation outlining how and when to use this function.

**Review by wesleykirklandsg — CHANGES_REQUESTED** [review]
Needs for documentation before publishing

File: README.md (hunk 2 above)
Comment (borjalopez-sg) [reply README.md]: I added it to the example.

File: codebuild-terraform.tf (hunk 1 above)
Comment (borjalopez-sg) [reply codebuild-terraform.tf:26]: I followed the original format. We can open a new PR with all the style changes.

**Review by JoseMotto-sg — APPROVED** [review]

**Review by ivancasco-sg — APPROVED** [review]

**Pull request merged**